hazelcast-hibernate5/src/main/java/com/hazelcast/hibernate/region/CacheKeyImpl.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (c) 2008-2016, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.hibernate.region;
+
+import java.io.IOException;
+
+import com.hazelcast.nio.ObjectDataInput;
+import com.hazelcast.nio.ObjectDataOutput;
+import com.hazelcast.nio.serialization.DataSerializable;
+import org.hibernate.engine.spi.SessionFactoryImplementor;
+import org.hibernate.type.Type;
+
+/**
+ * Cache key implementation
+ */
+public final class CacheKeyImpl implements DataSerializable {
+
+  private Object id;
+  private Type type;
+  private String entityOrRoleName;
+  private String tenantId;
+  private int hashCode;
```

Comment:
I thought we didn't need it. After merging this is, I will remove this class and `NaturalIdCacheKey` from 5.2 code so Maven shade plugin will enable us to use this version for both Hibernate 5.0.x and 5.2.x.